### DISCUSSION TITLE
STM32 Timer1 cc interrupt

### DISCUSSION THREAD

**Discussion by Gabschgi** [zephyrproject-rtos/zephyr#52555] [Q&A] [answered]
Hey,
beginner here. I have a question regarding the stm nucleo u575.
I wan't to use timer1 as a qdec and trigger an interrupt when a certain value is reached.
For this, I created two child nodes in the dt, one for the qdec (st,stm32-qdec) and a counter (st-stm32-counter).

I access the qdec node via the sensor struct. However I cannot get an interrupt running for the qdec. Then i tried to add a counter, where a interrupt should be triggered when a certain value is reached (i used the counter_set_channel_alarm() function to this).
The problem is, that the "up" interrupt triggers, an not the "cc".

I triet to implement a counter interrupt for timer3 where everything works fine. I noticed that every timer only has a global interrupt line, except timer1 (which as "brk", "up", "cc" and on other i cannot remember).
I looked into the counter implementation for the stm counter (zephyr/drivers/counter/counter_ll_stm32_rtc.c) and noticed that at the bottom at rtc_stm32_irq_config() always IRQn(0) is chosen. Could this be the problem or is this done on purpose? Is there any other way to implement a interrupt for a qdec? Did i miss something in the documentation?

Any help is appreciated :)

**Comment by FRASTM** [comment, marked as answer]
You could also have counter with  stm32 GP Timer : see  as examples drivers: counter: add st,stm32-counter to stm32h7 #52520  or https://github.com/zephyrproject-rtos/zephyr/pull/52583 for L5 counter with RTC
This is not in the device tree of the stm32U5, but could be adapted for tim2,3,4,5.
(I had something around that : https://github.com/FRASTM/zephyr/tree/counter_u5)
